pages/[id].js

```diff
@@ -1,27 +1,36 @@
 import { useRouter } from "next/router";
 import OutivityDetail from "@/components/OutivityDetail";
+import useSWR from "swr";
+import useFavorites from "@/lib/useFavorites";
 
-export default function OutivityDetailsPage({
-  outivities,
-  onDeleteOutivity,
-  favorites,
-  onToggleFavorite,
-}) {
+export default function OutivityDetailsPage() {
   const router = useRouter();
   const { id } = router.query;
-  const outivity = outivities.find((outivity) => outivity.id === id);
 
-  if (!outivity) {
+  const { data: outivity, isLoading } = useSWR(`/api/outivities/${id}`);
+  const { favorites, toggleFavorite } = useFavorites();
+
+  if (isLoading) {
     return <p>Loading...</p>;
   }
 
+  async function handleDeleteOutivity() {
+    const response = await fetch(`/api/outivities/${id}`, { method: "DELETE" });
+
+    if (!response.ok) {
+      console.log(response.status);
+      return;
```

Comment:
What if the response.ok is false? You make a console.log but the user is neither informed nor pushed to the homepage.

Reply:
Thanks for taking your time to review our code @Roland-Hufnagel 🙌 

We made some updates here, thanks for the hint!